src/main/java/org/radarcns/management/web/rest/OAuthClientsResource.java

```diff
@@ -352,7 +331,7 @@ private void checkProtected(ClientDetails details) {
         Map<String, Object> info = details.getAdditionalInformation();
         if (Objects.nonNull(info) && info.containsKey(PROTECTED_KEY)
                 && info.get(PROTECTED_KEY).toString().equalsIgnoreCase("true")) {
-            throw new CustomParameterizedException("Modification of a protected OAuth client is "
+            throw new AccessDeniedException("Modification of a protected OAuth client is "
```

Comment:
I think `InvalidAccessException` is more suitable than `AccessDeniedException` for this case.

Reply:
I'm not seeing an `InvalidAccessException` class in our project or in Spring, is it from another library? You're right, `AccessDenied` might create the expectation that there is a specific role that would allow this operation, which is not the case. So perhaps i'll just revert this change. This would then again yield a 400 Bad request, indicating that there is a problem with the request, not with the permissions. What do you think?

Reply:
Yes. `400` makes more sense. AccessDenied is more suitable for 401 i would say. Yes. I think it is not available in standard packages. But makes more sense. We could create the `InvalidAccessException` or `InvalidActivityException`. So i guess custom exception if necessary and status code `400` is good to go.